[Nithvin Leelakrishnan] iP

NightlyAffair opened the pull request:
CherylBot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Ever wanted an assistant but couldn't afford to hire one? Worry not CherylBot is here!

- text-based
- easy to learn 
- time ~consuming~ saving

All you need to is,
1. Download from [here.](https://github.com/NightlyAffair/ip)
2. double-click it.
3. add your tasks.
4. Let it manage your tasks for you 😉

And you can do all of this for **FREE!**

Features:

- [X] Managing Tasks
- [ ] Managing Deadlines

To run the program simply type 
`java jar Cheryl.jar`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

NightlyAffair committed:
Level 0

NightlyAffair committed:
Level 3

NightlyAffair committed:
level 3

NightlyAffair committed:
Level 5

NightlyAffair committed:
A-TextUiTesting

NightlyAffair committed:
Updated tests

NightlyAffair committed:
Enums and Exceptions added. Tests not updated.

NightlyAffair committed:
Update testcases to test for exceptions thrown to ensure undesirable behaviour is caught.

NightlyAffair committed:
Implement save features including read and write from data directory. Tasks are serialized upon writing and deserialized upon reading.

NightlyAffair committed:
Add date and time recoginition thorugh LocalDateTime variable.

NightlyAffair committed:
Abstract and encapsulate classes

Create a run method in Cheryl to reduce code in main. Create new UI class to handle interactions with user. Create new TaskList class to handle the list operations. Parser class to handle understanding user input.

NightlyAffair committed:
Package all files

Create 5 different packages namely the main package cheryl, followed by the sub packages: data,exception,task and ui. Test to ensure pacakges are imported correctly in all code.

NightlyAffair committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

NightlyAffair committed:
Set up Gradle in project

NightlyAffair committed:
First testing

NightlyAffair committed:
Add J-Unit Tests

Create J-Unit tests for tasks to ensure core code always stays intact. Ensure interactions with manager always stays intact.

NightlyAffair committed:
Add tests for robust testing

NightlyAffair committed:
Implement search function

Allow user to search all task description in the task list and retrieve it for them to view. Simple loop used to implement this to look through all tasks in the list to compile a list of those with the descriptions.

NightlyAffair committed:
Add Javadoc comments for all classes

NightlyAffair committed:
Follow java coding standard

NightlyAffair committed:
Merge tag 'A-JavaDoc'

* tag 'A-JavaDoc':
  Add Javadoc comments for all classes

# Conflicts:
#	src/main/java/cheryl/FileSystem.java

NightlyAffair committed:
Merge tag 'Level-9'

* tag 'Level-9':
  Implement search function

# Conflicts:
#	src/main/java/cheryl/util/TaskList.java

NightlyAffair committed:
Merge tag 'A-JUnit' of https://github.com/NightlyAffair/ip into A-JUnit

NightlyAffair committed:
Merge branch 'A-JUnit'

NightlyAffair committed:
Add CheckStyle and spotless

There are multiple ways to do this but the objective was to keep this automated as possible to reduce resources spend on styling. Hence I integrate spotless into gradle and checkstyle to automate both checking and fixing the formatting issues.

Parrycys commented on src/main/java/cheryl/task/Deadline.java:
```diff
@@ -0,0 +1,109 @@
+     * @return A new deadline task marked as completed.
+     */
+    @Override
+    public Task mark() {
```
Perhaps the mark and unmark methods could be under TaskList as the class handles the operations of the tasks ?

NightlyAffair committed:
First attempt at implementing JavaFX

Issue of main launcher not loading when running gradlew run. Builds fine with gradlew build.

Parrycys commented on src/main/java/cheryl/util/FileSystem.java:
```diff
@@ -0,0 +1,130 @@
+import cheryl.exception.FileCorruptedException;
+import cheryl.task.*;
+
+import java.io.*;
```
Should the implicit imports be removed to keep the imported classes explicitly ? 😺

Parrycys commented on src/main/java/cheryl/util/TaskList.java:
```diff
@@ -0,0 +1,174 @@
+     *
+     * @return The list of tasks managed by the TaskList.
+     */
+    public ArrayList<Task> get() {
```
Perhaps the name could be getTaskList() so that when we call .get() we will not mistake it as the ArrayList get method ?

NightlyAffair committed:
No changes

Parrycys commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,22 @@
-
+Hello! I'm Cheryl
+What can I do for you?
+Say help if you don't know what you are doing
```
I like how you added this line to provided extra help if the user is new 👍

Parrycys commented on src/main/java/cheryl/ui/TimeProcessor.java:
```diff
@@ -0,0 +1,37 @@
+            }
+
+        } catch (DateTimeParseException e) {
+            System.out.println("Invalid date format: " + dateTime + "\n please provide date and time again in the following format: 2/12/2019 1800");
```
Should this line be indented as this line is longer than the hard limit of 120 characters ?

Parrycys commented on src/main/java/cheryl/ui/UI.java:
```diff
@@ -0,0 +1,36 @@
+        String sayMark = "Say mark 1 to mark task 1 in the list as done";
+        String sayUnMark = "Say unmark 1 to unmark task 1 in the list as done";
+        String sayDelete = "Say delete 1 to delete task 1 in the list";
+        return sayTodo + '\n' + sayDeadlines + '\n' + sayEvent + '\n' + sayList + '\n' + sayMark + '\n' + sayUnMark + '\n' + sayDelete;
```
Should this line be indented as well as this line is longer than 120 characters ?

Parrycys reviewed (commented):
LGTM. Just some nits to fix. I noticed some lines are not indented accordingly to the coding standard. The rest are well done ! 😄

nathtiekk commented on src/main/java/Cheryl.java:
```diff
@@ -0,0 +1,50 @@
+     */
+    public void run() {
+        UI.printIntro();
+        Boolean runMainLoop = true;
```
Hello! A small nitpick here is that the coding standard recommends that Boolean variables should be named to sound like booleans: sound like a question.
```suggestion
        Boolean shouldRunMainLoop = true;
```

nathtiekk commented on src/main/java/cheryl/task/Deadline.java:
```diff
@@ -0,0 +1,109 @@
+    @Override
+    public String serialize() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+        return TaskType.DEADLINE + "||" + this.check + "||" + this.description + "||" + this.date.format(formatter);
```
For a line of code that is long, perhaps it would be more readable to split that single line into a new line, then indent the new line by 8 spaces as per the coding standard recommendation.
```suggestion
        return TaskType.DEADLINE + "||" + this.check + "||" 
            + this.description + "||" + this.date.format(formatter);
```

nathtiekk commented on src/main/java/cheryl/task/Event.java:
```diff
@@ -0,0 +1,138 @@
+        this.endDate = endDate;
+    }
+
+    /**
+     * Returns a string representation of the event task, including its type, status, description,
+     * start date, and end date.
+     *
+     * @return A string in the format "[E][status] Task description (from: startDate to: endDate)".
+     */
```
I like your detailed and clear JavaDoc comments here, and in many parts of your code!

nathtiekk commented on src/main/java/cheryl/ui/UserCommand.java:
```diff
@@ -0,0 +1,64 @@
+            return "Huh?? Please enter a valid input...";
+        }
+
+        switch (command) {
+            case HELP:
+                return UI.helpString();
+            case LIST:
+                return taskList.listOut();
+        }
```
I believe the switch statements that is not Lambda-style should be in this format:
```suggestion
        switch (command) {
        case HELP:
            return UI.helpString();
            // Fallthrough
        case LIST:
            return taskList.listOut();
            // Fallthrough
        }
```
Note the lack of indentation in the case statements, and the "break;" or "// Fallthrough" within every case statement as per the coding standard guidelines

nathtiekk reviewed (commented):
Overall, I like your file management where you put all related classes in the same package; and your code LGTM with some minor nits.

NightlyAffair committed:
Create GUI for Cheryl

Use JavaFX to create a GUI, following the tutorial.

NightlyAffair committed:
no message

NightlyAffair committed:
Merge tag 'A-CheckStyle'

* tag 'A-CheckStyle':
  Add CheckStyle and spotless

# Conflicts:
#	src/main/java/cheryl/main/Cheryl.java
#	src/main/java/cheryl/ui/UI.java

NightlyAffair committed:
Clean up code and name variables better

NightlyAffair committed:
Add assertions

NightlyAffair committed:
Merge branch 'master' into A-Assertions

* master:
  Clean up code and name variables better

NightlyAffair committed:
no message

NightlyAffair committed:
Create Managers to handle different types of data

NightlyAffair committed:
no message

NightlyAffair committed:
Implement logic for CLI functionality

Have implemented Java logic to move through menus for CLI version but now need to change GUI from javaFX to catch up to said changes.

NightlyAffair committed:
Implement main menu system for javaFX input

Working menu system to transition between multiple managers to handle different things.

NightlyAffair committed:
Implement FileSystem however buggy

NightlyAffair committed:
Add elegant error handling methods

NightlyAffair committed:
Add contact functionality

NightlyAffair committed:
Add functionality to add, remove, edit and list out contacts

NightlyAffair committed:
Bug fixing

NightlyAffair committed:
Fix all bugs

NightlyAffair committed:
Allow for GUI scaling

NightlyAffair committed:
Add README.md and fix bugs with editing contacts

NightlyAffair committed:
Improve Code Quality

NightlyAffair committed:
Improve error handling and ensure 2 J-Unit tests are functional

NightlyAffair committed:
Update README.md

NightlyAffair committed:
Update README.md
